clojure/src/info/lotharschulz/armstrong_numbers/armstrong_numbers.clj

```diff
@@ -0,0 +1,38 @@
+(ns info.lotharschulz.armstrong_numbers.armstrong_numbers
+  (:import [java.Math]))
+
+(defn get-number-of-digits [^Integer n]
+  {:pre [(not (nil? n)) (integer? n)]}
+  "Returns the :getNumberOfDigits of n if n is an integer and greater than -1"
+  (if (zero? n)
+    1
+    (if (pos? n )
+      (+ (int (Math/log10 n )) 1)
+      nil)))
+
+(defn get-digits [^Integer n]
+  {:pre [(not (nil? n)) (integer? n)]}
+  "return number of digits of given int n"
+    (if (zero? n)
+      [0]
+      (if (pos? n)
+        (into [] (map #(Character/getNumericValue %) (str n)))
+        [])))
+
+(defn is-armstrong-number [^Integer n]
+  {:pre [(not (nil? n)) (integer? n)]}
+  "Returns true if the given number bigger than -1 is an armstrong number"
+  (if (zero? n)
+    false
+    (if (pos? n)
+      (= n (reduce + (map #(int (Math/pow % (get-number-of-digits n))) (get-digits n))))
+      false)))
```

Comment:
a cleaner implementation might look like:

```clojure
(defn armstrong-number? [n]
  (and (pos? n)
       (let [digits-of-n      (digits n)
             number-of-digits (count digits-of-n)]
         (== n (->> digits-of-n
                    (map #(Math/pow % number-of-digits))
                    (reduce +))))))
```

* `->>` makes nested expression look not so nested
* `==` compares numbers, `(== 1 1.0) => true`, but `(= 1 1.0) => false`
* it actually does not hurt to call it with negative numbers, but for purity we can add `(pos? n)`